src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -203,10 +203,29 @@ void evaluate_impl(
       "e.g. evaluate<ti::a, ti::b>(L, R(ti::b, ti::a));, where R's first "
       "index has 2 spatial dimensions but L's second index has 3 spatial "
       "dimensions. Check RHS and LHS indices that use the same generic index.");
+  static_assert(Derived::height_relative_to_closest_tensor_leaf_in_subtree <
+                    std::numeric_limits<size_t>::max(),
+                "This either indicates that no Tensors were found in the RHS "
```

Comment:
"This" is not a great word in general, since it makes reading more difficult because of the implied context. Here, I don't understand what "this" refers too. What about `Either no Tensors were found...` and then `maximum size_t value (extremely unlikely)`